sympy/assumptions/sathandlers.py

```diff
@@ -186,44 +186,34 @@ def _eval_apply(self, expr, pred):
         # return And(*[Or(*map(Not, c)) for c in combinations(pred_args, 2)]) & Or(*pred_args)
 
 
+_old_assump_getters = {
+    Q.positive: lambda o: o.is_positive,
+    Q.zero: lambda o: o.is_zero,
+    Q.negative: lambda o: o.is_negative,
+    Q.nonpositive: lambda o: o.is_nonpositive,
+    Q.nonzero: lambda o: o.is_nonzero,
+    Q.nonnegative: lambda o: o.is_nonnegative,
+    Q.rational: lambda o: o.is_rational,
+    Q.irrational: lambda o: o.is_irrational,
+    Q.even: lambda o: o.is_even,
+    Q.odd: lambda o: o.is_odd,
+    Q.integer: lambda o: o.is_integer,
+    Q.composite: lambda o: o.is_composite,
+    Q.imaginary: lambda o: o.is_imaginary,
+    Q.commutative: lambda o: o.is_commutative,
+}
+
 def _old_assump_replacer(obj):
     if not isinstance(obj, AppliedPredicate):
         return obj
```

Comment:
Why does this need to check if `obj` is an `AppliedPredicate`?

Reply:
It seems this check can be removed.